src/main/java/ListNode.java

```diff
@@ -0,0 +1,44 @@
+public class ListNode {
+    public ListNode next;
+    public int data;
+
+    public ListNode(int data) {
+        this.data = data;
```

Comment:
It's a good habit to set your next pointer to null here so that you're explicit about setting all the values in the type.